From 0fb9013aa056db642457e93a20499fd9b46ba436 Mon Sep 17 00:00:00 2001
From: Kam Nasim <kam.nasim@windriver.com>
Date: Mon, 22 Jan 2018 11:18:08 -0500
Subject: [PATCH] CGTS-8701: Remove the Keystone-admin app

Following the Pike rebase, no services are using Identity V2 and
therefore we can shut off the Keystone admin port / app, as in Identity
V3 the public endpoint and admin endpoints both offer identical services
---
 lib/puppet/provider/keystone.rb | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/lib/puppet/provider/keystone.rb b/lib/puppet/provider/keystone.rb
index 8eb171d..3c310dc 100644
--- a/lib/puppet/provider/keystone.rb
+++ b/lib/puppet/provider/keystone.rb
@@ -171,12 +171,16 @@ class Puppet::Provider::Keystone < Puppet::Provider::Openstack
   end
 
   def self.get_admin_endpoint
+    # NOTE (knasim-wrs): As of the Pike rebase, the public port(5000)
+    # provides the same functionality as the admin port(35357). We
+    # shall therefore not deploy the keystone-admin app and return
+    # the public port
     endpoint = nil
     if keystone_file
       if url = get_section('DEFAULT', 'admin_endpoint')
         endpoint = url.chomp('/')
       else
-        admin_port = get_section('DEFAULT', 'admin_port') || '35357'
+        admin_port = get_section('DEFAULT', 'public_port') || '5000'
         host = clean_host(get_section('DEFAULT', 'admin_bind_host'))
         protocol = ssl? ? 'https' : 'http'
         endpoint = "#{protocol}://#{host}:#{admin_port}"
-- 
1.8.3.1

